.github/workflows/Pre_release.yml

```
@@ -12,29 +20,25 @@ jobs:
   release:
     name: "Publish Pre-Release"
     runs-on: ubuntu-latest
+    permissions:
+      contents: write
     steps:
       - uses: actions/checkout@v4
         with:
-          fetch-depth: 0  # Needed determine latest tag
+          fetch-depth: 0  # Needed determine tags
           persist-credentials: true
-          ref: main
-      - name: Get Latest Tag
-        id: latest_tag
-        uses: WyriHaximus/github-action-get-previous-tag@v1
-        # TODO: in future we might need to perform backports, at which point we'll need to check
-        # that the project.version has not already been tagged 
-      - name: Check for updated Project version
+          ref: ${{ inputs.commit }}
+      - name: Check for new Project version
         id: project_toml
         shell: julia --color=yes --project {0}
         env:
           LATEST_TAG: ${{ steps.latest_tag.outputs.tag }}
         run: |
           using Pkg.Types
           project = read_project("Project.toml")
-          latest_tag = parse(VersionNumber, ENV["LATEST_TAG"])
+          tags = parse.(VersionNumber, split(readchomp(`git tag -l`), "\n"))
```

Just curious if you verified this? I thought maybe we'd have to do `git fetch --tags`

---

yup it seems to work https://github.com/beacon-biosignals/Ray.jl/actions/runs/6630217614/job/18011182410?pr=220#step:3:18

---

If we create a tag that isn't a version number this will break. We should do:
```suggestion
          tags = split(readchomp(`git tag -l`), "\n")
```
and do a string comparison later